docs/user-manuals/persistentpodstate.md

```diff
@@ -0,0 +1,156 @@
+---
+title: PersistentPodState
+---
+
+**FEATURE STATE:** Kruise v1.2.0
+
+In Kubernetes, Pod is the "smallest unit of immutable infrastructure", and any change in configuration or environment (e.g., business release, node exception eviction, machine vacations, etc.) will cause the Pod to be rebuilt.
+In turn, the state associated with the Pod deployment will change, such as the Node machine it belongs to, the Pod IP, the local PV, etc. The above phenomenon is an acceptable behavior for most stateless applications.
+
+However, with the development of cloud-native, more and more companies have started to deploy stateful applications to K8S, based on the objective fact that there are technical differences among companies and the urgent desire to enjoy the dividends brought by K8S as soon as possible.
+Many of the stateful services migrated to K8S are not completely cloud-native transformation, and still retain many of the deployment features of the physical machine era, such as:
+
+- **Service discovery Middleware services are exceptionally sensitive to the Pod IP after deployment, requiring that the IP cannot be changed.**
+
+- **Database related services persist data to the host disk, and changes to the Node to which they belong will result in data loss.**
+
+For such stateful services as described above, which have a strong dependency on Pod deployment state, Kruise is able to maintain the state of Pods after their first deployment to some extent by PersistentPodState CRD, e.g., "Fixed IP Scheduling".
+
+## API Definition
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: PersistentPodState
+metadata:
+  name: echoserver
+  namespace: echoserver
+spec:
+  targetRef:
+    # Native k8s or kruise StatefulSet
+    # only support StatefulSet
+    apiVersion: apps.kruise.io/v1beta1
+    kind: StatefulSet
+    name: echoserver
+  # required node affinity. As follows, Pod rebuild will force deployment to the same Zone
+  requiredPersistentTopology:
+    nodeTopologyKeys:
+      failure-domain.beta.kubernetes.io/zone[,other node labels]
+  # preferred node affinity. As follows, Pod rebuild will preferred deployment to the same node
+  preferredPersistentTopology:
+    - preference:
+        nodeTopologyKeys:
+          kubernetes.io/hostname[,other node labels]
+      weight: 100
+```
+
+## Implementation
+PersistentPodState implementation principle is as follows, for detailed design please refer to: [PPS Proposal](https://github.com/openkruise/kruise/blob/master/docs/proposals/20220421-persistent-pod-state.md).
+
+![PersistentPodState](/img/docs/user-manuals/persistent-pod-state.png)
```

Comment:
We shall start with how to use PPS directly, then introduce the helper annotation in statefulset for ip retention